src/logics/ArithLogic.cc

```diff
@@ -0,0 +1,1161 @@
+    , sym_Real_PLUS(declareFun_Commutative_NoScoping_LeftAssoc(tk_real_plus, sort_REAL, {sort_REAL, sort_REAL}))
+    , sym_Real_TIMES(declareFun_Commutative_NoScoping_LeftAssoc(tk_real_times, sort_REAL, {sort_REAL, sort_REAL}))
+    , sym_Real_DIV(declareFun_NoScoping_LeftAssoc(tk_real_div, sort_REAL, {sort_REAL, sort_REAL}))
+    , sym_Real_EQ(sortToEquality[sort_REAL])
+    , sym_Real_LEQ(declareFun_NoScoping_Chainable(tk_real_leq, sort_BOOL, {sort_REAL, sort_REAL}))
+    , sym_Real_LT(declareFun_NoScoping_Chainable(tk_real_lt, sort_BOOL, {sort_REAL, sort_REAL}))
+    , sym_Real_GEQ(declareFun_NoScoping_Chainable(tk_real_geq, sort_BOOL, {sort_REAL, sort_REAL}))
+    , sym_Real_GT(declareFun_NoScoping_Chainable(tk_real_gt, sort_BOOL, {sort_REAL, sort_REAL}))
+    , sym_Real_ITE(sortToIte[sort_REAL])
+    , sym_Real_DISTINCT(sortToDisequality[sort_REAL])
+
+    , sort_INT(declareSortAndCreateFunctions(s_sort_int))
+    , term_Int_ZERO(mkConst(sort_INT, tk_int_zero))
+    , term_Int_ONE(mkConst(sort_INT, tk_int_one.c_str()))
+    , term_Int_MINUSONE(mkConst(sort_INT, tk_int_minus + tk_int_one))
+    , sym_Int_ZERO(getSymRef(term_Int_ZERO))
+    , sym_Int_ONE(getSymRef(term_Int_ONE))
+    , sym_Int_NEG(declareFun_NoScoping(tk_int_neg, sort_INT, {sort_INT}))
+    , sym_Int_MINUS(declareFun_NoScoping_LeftAssoc(tk_int_minus, sort_INT, {sort_INT, sort_INT}))
+    , sym_Int_PLUS(declareFun_Commutative_NoScoping_LeftAssoc(tk_int_plus, sort_INT, {sort_INT, sort_INT}))
+    , sym_Int_TIMES(declareFun_Commutative_NoScoping_LeftAssoc(tk_int_times, sort_INT, {sort_INT, sort_INT}))
+    , sym_Int_DIV(declareFun_NoScoping_LeftAssoc(tk_int_div, sort_INT, {sort_INT, sort_INT}))
+    , sym_Int_MOD(declareFun_NoScoping(tk_int_mod, sort_INT, {sort_INT, sort_INT}))
+    , sym_Int_EQ(sortToEquality[sort_INT])
+    , sym_Int_LEQ(declareFun_NoScoping_Chainable(tk_int_leq, sort_BOOL, {sort_INT, sort_INT}))
+    , sym_Int_LT(declareFun_NoScoping_Chainable(tk_int_lt, sort_BOOL, {sort_INT, sort_INT}))
+    , sym_Int_GEQ(declareFun_NoScoping_Chainable(tk_int_geq, sort_BOOL, {sort_INT, sort_INT}))
+    , sym_Int_GT(declareFun_NoScoping_Chainable(tk_int_gt, sort_BOOL, {sort_INT, sort_INT}))
+    , sym_Int_ITE(sortToIte[sort_INT])
+    , sym_Int_DISTINCT(sortToDisequality[sort_INT])
+
+    , arithType(arithType)
+{
+    assert(static_cast<int>(logicNames.size()) == logicTypes.size());
+}
+
+bool ArithLogic::hasIntegers() const {
+    switch (arithType) {
+        case ArithType::LIA:
+        case ArithType::NIA:
+        case ArithType::IDL:
+        case ArithType::LIRA:
+        case ArithType::NIRA:
+            return true;
+        default:
+            return false;
+    }
+}
+
+bool ArithLogic::hasReals() const {
+    switch (arithType) {
+        case ArithType::LRA:
+        case ArithType::NRA:
+        case ArithType::RDL:
+        case ArithType::LIRA:
+        case ArithType::NIRA:
+            return true;
+        default:
+            return false;
+    }
+}
```

Comment:
I sill believe we could do this with `Logic_t` and do not need to introduce a new enum for this.

Reply:
Let's discuss the best way in a new PR.